src/main/java/com/ecommerce/ECommerceApp/model/Order.java

```diff
@@ -100,7 +101,7 @@ public float getTotalAmount() {
      *
      * @param totalAmount the total amount of the order
      */
-    public void setTotalAmount(float totalAmount) {
+    public void setTotalAmount(BigDecimal totalAmount) {
```

Comment:
Replaced float with BigDecimal for monetary values to prevent rounding issues in financial calculations, ensuring precision and reliability. As Martin Fowler discusses in ["Technical Debt"](https://martinfowler.com/bliki/TechnicalDebt.html), addressing data accuracy early on reduces technical debt and avoids potential issues down the line.